privacy_security.qmd

```diff
@@ -1,81 +1,1064 @@
+
+FL aims to train machine learning models across decentralized networks of devices or systems while keeping all training data localized. In FL, each participating device leverages its local data to calculate model updates which are then aggregated to build an improved global model. However, the raw training data itself is never directly shared, transferred, or compiled together. This privacy-preserving approach allows jointly developing ML models without centralizing the potentially sensitive training data in one place.
+
+The figure below from [@MAL-083] illustrates the FL lifecycle. The training data always remains on the client data, the model repeatedly is sent back and forth between individual devices and server for local updates and compiling the global model, respectively.
+
+![](images/security_privacy/image7.png)
+
+One of the most common model aggregation algorithms is Federated Averaging (FedAvg) where the global model is created by averaging all of the parameters from local parameters. While FedAvg works well with independent and identically distributed data (IID), alternate algorithms like Federated Proximal (FedProx) are crucial in real-world applications where data is often non-IID. FedProx is designed for the FL process when there is significant heterogeneity in the client updates due to diverse data distributions across devices, computational capabilities, or varied amounts of data.
+
+By leaving the raw data distributed and exchanging only temporary model updates, federated learning provides a more secure and privacy-enhancing alternative to traditional centralized machine learning pipelines. This allows organizations and users to collaboratively benefit from shared models while maintaining control and ownership over their sensitive data. The decentralized nature of FL also makes it robust to single points of failure.
+
+Imagine a group of hospitals that want to collaborate on a study to predict patient outcomes based on their symptoms. However, due to privacy concerns and regulations like HIPAA, they cannot share their patient data with each other. Here's how Federated Learning can help.
+
+- **Local Training:** Each hospital trains a machine learning model onits own patient data. This training happens locally, meaning thedata never leaves the hospital's servers.
+
+- **Model Sharing:** After training, each hospital only sends themodel (specifically, the parameters or weights of the model) to acentral server. They do not send any patient data.
+
+- **Aggregating Models:** The central server aggregates these modelsfrom all hospitals into a single, more robust model. This processtypically involves averaging the model parameters.
+
+- **Benefit:** The end result is a machine learning model that haslearned from a wide range of patient data without any of thatsensitive data having to be shared or leave its original location.
+
+#### Trade-offs
```

Comment:
Might be worth calling out that FL is still vulnerable to some types of attacks like data poisoning?